### ISSUE TITLE
Make upload_to_testflight FastlaneCore::Interface::FastlaneError messages useful

### ISSUE THREAD

**Issue by xster** [fastlane/fastlane#11902]
The exception message just says `Error uploading ipa file, for more information see above` which makes it hard to do any specific error handling.

**Comment by xster** [comment]
Seems like even reassigning $stdout or $stderr can't get the actual output from upload_to_testflight

**Comment by taquitos** [comment]
You can add the `--verbose` flag and that should give you a little more info.

**Comment by cpk1** [comment]
I think the issue xster is talking about is using "exception.message" in the fastfile will give you the error he is talking about. If you have verbose turned on you can go look at the logs but it is nice to provide a meaningful message when messaging/emailing a group.

**Comment by xster** [comment]
ya, I meant being able to programmatically read out the output of upload_to_testflight in Fastfile and control flowing accordingly.

**Comment by taquitos** [comment]
Ahh, that sounds like a good idea. Is that something you'd want to help with? If so, we'd be happy to review any PR 🚀

**Comment by cpk1** [comment]
👍 I'll try to take a stab at it. Although I'm not sure how upload_to_play_store does it :)

**Comment by cpk1** [comment]
So I think I have something that provides better output from what happened during transporter... Not sure if you want me to just wrap it up and send a PR or have some discussion to make sure the way I'm doing this is OK with the team. LMK @taquitos

**Comment by taquitos** [comment]
Yeah, it would be awesome to get your thoughts. If it's easier for you to throw together a PR for us to review, or if you want to write out your ideas, that also works for us. Totally up to you as we're happy to review either 👍

**Comment by cpk1** [comment]
What I've got working locally right now is using the error parser in itunes_transporter ( https://github.com/cpk1/fastlane/blob/cpk1-transporterLogging/fastlane_core/lib/fastlane_core/itunes_transporter.rb ) to add errors that we want to return to a new instance variable and then returning that variable in addition to the transporter exit status.

In pilot I've changed UI.user_error to print out the new variable that is being returned from itunes_transporter: https://github.com/cpk1/fastlane/blob/cpk1-transporterLogging/pilot/lib/pilot/build_manager.rb#L37-L43

Doing this prints "Error uploading ipa file:
[Transporter Error Output]: ERROR ITMS-90189: Redundant Binary Upload. You've already uploaded a build with build number '31' for version number '1.0'. Make sure you increment the build string before you upload your app to iTunes Connect. Learn more in Xcode Help (http://help.apple.com/xcode/mac/current/#/devba7f53ad4)." from "exception.to_s" in the slack plugin

I see doing this (or something similar) for errors super useful because it could be leveraged to send emails based on type of failure or even rebuild if it is simply a build number that needs to be increased. Similarly for warnings we could chose which warnings we want to ignore and which warnings we should actually alert for action that needs to be taken.

However I understand that this work here is very focused on solving the specific problem of returning more actionable errors from transporter and I'm not sure if this is the appropriate way to do this. More than happy to tackle this from a different angle if that's what the team would prefer, I figure this is a good start to at least understand what I'm looking to be able to get returned from itunes_transporter.
